### ISSUE TITLE
XCall solidity protocol fee is not enforced

### ISSUE THREAD

**Issue by AntonAndell** [icon-project/xcall-multi#273]
Protocol fee is not enforced, currently no error is raised if the user does opt not to pay it

**Comment by gcranju** [comment]
About the protocol fee in solidity, we have there, called from line 182
```
    function claimProtocolFee() internal {
        uint256 balance = address(this).balance;
        require(balance >= protocolFee, "InsufficientBalance");
        feeHandler.transfer(balance);
    }
```

this will revert if the protocol fee is not passed, right?

**Comment by AntonAndell** [comment]
Yeah this seems to have been resolved in xCall 2.1
